geowebcache/tms/src/main/java/org/geowebcache/service/tms/TMSService.java

```diff
@@ -178,10 +178,29 @@ public ConveyorTile getConveyor(HttpServletRequest request, HttpServletResponse
         }
 
         String[] yExt = params[paramsLength - 1].split("\\.");
+        String x = params[paramsLength - 2];
+        String y = yExt[0];
+        String z = params[paramsLength - 3];
+
+        // look for flipY parameter and do flipping if flipY=true
+        String flip = yExt[1].toUpperCase();
+        if (flip.contains("?FLIPY=TRUE")) {
```

Comment:
This seems a fragile way to handle query parameters. What if for example one would make a request like 
"...?cacheBreaker=randomeNumber&flipy=true"?
Instead, I would throw away anything that is beyond the ? and use ``getParameterNames`` and ``getParameter`` to grab the eventual flipy parameter.